### DISCUSSION TITLE
CEFSharp won't play my uploaded avi, ps9, or ps8 encoded Youtube videos, but will play many other videos on Youtube.

### DISCUSSION THREAD

**Discussion by BillBogart** [cefsharp/CefSharp#4834] [Q&A]
I am trying to get customer Youtube videos to play in Chromium ver. 115.3.130.  I have used HandBrake tool to encode videos in AV1 (SVT), AV1 10-bit (SVT),  VP9, and VP9 10-bit   None of them will play in Chromium CefSharp.MinimalExample.WinForms.net472 app.  YouTube videos created by other people play correctly in CefSharp.MinimalExample.WinForms.net472 app, but none of my landscape videos will play. I get the "Your browser can't play this video" message. I was able to get a short from MOV format to encode AV1 (SVT) and play from the same youtube account channel,  but none of the non short videos will play.   

Any help would be greatly appreciated.

I get these generate_204 messages:
/watch?v=WxLSYoXfxtM:1 The resource https://i.ytimg.com/generate_204 was preloaded using link preload but not used within a few seconds from the window's load event. Please make sure it has an appropriate `as` value and it is preloaded intentionally.
/watch?v=WxLSYoXfxtM:1 The resource https://rr1---sn-ab5l6ndr.googlevideo.com/generate_204 was preloaded using link preload but not used within a few seconds from the window's load event. Please make sure it has an appropriate `as` value and it is preloaded intentionally.
/watch?v=WxLSYoXfxtM:1 The resource https://rr1---sn-ab5l6ndr.googlevideo.com/generate_204?conn2 was preloaded using link preload but not used within a few seconds from the window's load event. Please make sure it has an appropriate `as` value and it is preloaded intentionally.

**Comment by amaitland** [comment]
> I am trying to get customer Youtube videos to play in Chromium ver. 115.3.130

How does the latest version behave?

Only free audio/video codes are supported as per https://github.com/cefsharp/CefSharp/wiki/General-Usage#multimedia-audiovideo

Long as your audio and video codes are on the supported list then it should behave the same as in Chrome.

**Reply by BillBogart** [reply]
Turns out YouTube streams low view videos or unpopular videos using H.264.  Doesn't matter if you encode your videos using PV9 and then upload them to YouTube, YouTube will still encode them using using H.264 when streaming to client web browsers.  I have found a solution that tricks or forces YouTube into streaming your low view unpopular videos with VP9 which will play in Chromium.

**Reply by SLT-World** [reply]
Could you share your solution? @BillBogart, Thanks.

**Reply by BillBogart** [reply]
I used HandBrake utility for my encoding. The main thing is to upscale your video to at least 2048x1152 and encode using VP9. My steps using hand brake are:
1) Select WebM as the format.
2) Enable “Allow Upscaling.”
3) Enter 2048 and 1152 for Maximum size.
4) Enter 2048 and 1152 for Scaled size.
5) Select VP9 video encoder.
6) Select Peak Framerate.
7) Select “Vorbis” for audio encoder. 
8) run "Start Encode"  

This should produce a high enough resolution video that tricks Youtube into streaming with VP9.
